packages/client/src/app/auth/signup/page.tsx

```diff
@@ -1,39 +1,112 @@
-import React from 'react';
+'use client';
+
+import React, { useState } from 'react';
+import axios from 'axios';
 
 export default function SignupPage() {
+  const [formData, setFormData] = useState({
+    name: '',
+    email: '',
+    password: '',
+    role: 'student',
+  });
+
+  const [error, setError] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    setFormData({ ...formData, [e.target.name]: e.target.value });
+  };
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setError('');
+    setIsLoading(true);
+
+    try {
+      const response = await axios.post(
+        'http://localhost:5000/v1/auth/register',
```

Comment:
Maybe can extract out to API folder and follow the structure, we're using next as proxy and the port number may also change during deployment